src/components/WordGroupSelection/WordGroupSelection.js

```diff
@@ -0,0 +1,116 @@
+    const [peopleWords, setPeopleWords] = useState(null);
+    const [actionWords, setActionWords] = useState(null);
+    const [toyWords, setToyWords] = useState(null);
+    const [colorWords, setColorWords] = useState(null);
+    const [animalWords, setAnimalWords] = useState(null);
+    const [transportWords, setTransportWords] = useState(null);
+    const [bodyWords, setBodyWords] = useState(null);
+    const [clothingWords, setClothingWords] = useState(null);
+    const [foodWords, setFoodWords] = useState(null);
+    const [foodWords2, setFoodWords2] = useState(null);
+    const [furnitureWords, setFurnitureWords] = useState(null);
+    const [emotionWords, setEmotionWords] = useState(null);
+    const [mediaWords, setMediaWords] = useState(null);
+    const [animalWords2, setAnimalWords2] = useState(null);
+    const [animalWords3, setAnimalWords3] = useState(null);
+
+    collectedWordGroupsService.all().then(function(res) {
+        word_groups = res;
+        word_keys = Object.keys(word_groups);
+        setPeopleWords(word_groups[word_keys[0]])
+        setActionWords(word_groups[word_keys[1]])
+        setToyWords(word_groups[word_keys[2]])
+        setColorWords(word_groups[word_keys[3]])
+        setAnimalWords(word_groups[word_keys[4]])
+        setTransportWords(word_groups[word_keys[5]])
+        setBodyWords(word_groups[word_keys[6]])
+        setClothingWords(word_groups[word_keys[7]])
+        setFoodWords(word_groups[word_keys[8]])
+        setFoodWords2(word_groups[word_keys[9]])
+        setFurnitureWords(word_groups[word_keys[10]])
+        setEmotionWords(word_groups[word_keys[11]])
+        setMediaWords(word_groups[word_keys[12]])
+        setAnimalWords2(word_groups[word_keys[13]])
+        setAnimalWords3(word_groups[word_keys[14]])
+    });
+
+    const [clickedName, setClickedName] = useState('');
+    const [clickedGroup, setClickedGroup] = useState([]);
+    const [selectionMode, setSelectionMode] = useState(false);
+
+    function handleClick(group, groupName) {
+        setClickedGroup(group);
+        setClickedName(groupName);
+        setSelectionMode(!selectionMode);
+    }
+
+    function handleChangeSelectionMode(selectStatus) {
+        setSelectionMode(selectStatus);
+    }
+
+    return(
+        <div className="Background">
+            <div className="WordGroups">
+                <div onClick={() => handleClick(peopleWords, "People")}>
+                    <WordGroupIcon number={1} name={"People"}/>
+                </div>
+                <div onClick={() => handleClick(actionWords, "Action")}>
+                    <WordGroupIcon number={2} name={"Action"}/>
+                </div>
+                <div onClick={() => handleClick(toyWords, "Toy")}>
+                    <WordGroupIcon number={3} name={"Toy"}/>
+                </div>
+                <div onClick={() => handleClick(colorWords, "Colors")}>
+                    <WordGroupIcon number={4} name={"Colors"}/>
+                </div>
```

Comment:
Then if you don't have a bunch of state variables, on render you could iterate over the one state variable instead of manually rendering